server/src/resources/base/mod.rs

```diff
@@ -1,5 +1,68 @@
 //! Base
 
-// Expose grpc resources
-mod grpc;
-pub use grpc::*;
+use crate::common::ArrErr;
+use chrono::DateTime;
+use chrono::Datelike;
+use chrono::NaiveDateTime;
+use chrono::Timelike;
+use chrono::Utc;
+use prost_types::Timestamp;
+
+/// Validation Error
+#[derive(Debug)]
+pub struct ValidationError {
+    pub field: String,
+    pub error: String,
+}
+
+/// Convert a prost_types::Timestamp (used by grpc)
+/// into a chrono::DateTime::<Utc> (used by postgres)
+/// TODO: Can we use Traits for or simplify in another way?
+pub fn ts_to_dt(ts: &Timestamp) -> Result<DateTime<Utc>, ArrErr> {
+    let seconds = ts.seconds;
+    let nanos = match ts.nanos.try_into() {
```

Comment:
Trying to convert from pros_types::Timestamp into a std::time::SystemTime will give this error:
```
the trait bound `SystemTime: From<&prost_types::Timestamp>` is not satisfied
```

`NaiveDateTime::from_timestamp(seconds, nanos)` expects to get an `u32` for the `nanos` value while `prost_types::Timestamp::nanos` will return an `i32`.
The `try_into()` will know this since rust knows what value the `from_timestamp` function expects where the variable is used. I will add it as a type though, to make it clear it's a conversion to `u32` at that point.
`ts.nanos.into()` does not support `u32` directly, so `try_into` is needed for that.